packages/mdc-menu-surface/index.ts

```diff
@@ -0,0 +1,206 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import MDCComponent from '@material/base/component';
+import {SpecificEventListener} from '@material/dom/index';
+import {Corner, CornerBit, cssClasses, MenuDimensions, MenuDistance, MenuPoint, strings} from './constants';
+import {MDCMenuSurfaceFoundation} from './foundation';
+import * as util from './util';
+
+type RegisterFunction = () => void;
+
+class MDCMenuSurface extends MDCComponent<MDCMenuSurfaceFoundation> {
+  static attachTo(root: Element): MDCMenuSurface {
+    return new MDCMenuSurface(root);
+  }
+
+  anchorElement: Element | null = null;
+
+  protected root_!: HTMLElement; // assigned in MDCComponent constructor
+
+  private previousFocus_: HTMLElement | SVGElement | null = null;
```

Comment:
Is there a reason we're using `HTMLElement | SVGElement` for these rather than just `Element`?

Reply:
`Element` lacks `.focus()`, which we call in the adapter.